### PR TITLE
Implement 'generateThumbnail' method (closes #87)

### LINKED ISSUE #87: Implement API for generating thumbnails
Currently we have different code for generating screenshot thumbnails for macOS and Windows, tightly coupled with `screenshot` modules. The goal is:

1. Use the same code for generating thumbnails on different OSes;

2. Provide the thumbnails generating API in order to enable unified thumbnails generation for different browser providers on `testcafe` level.

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @VasilyStrelyaev @AlexanderMoskovkin @helen-dikareva

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
lgtm

File: src/api/generate-thumbnail.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * Creates a small preview image called thumbnail image for the specified image file.
+ * @function
+ * @async
+ * @name generateThumbnail
+ * @param {string} imagePath - Specifies the path to the source image, must be in PNG format.
+ * @param {string} thumbnailPath - Specifies the path to the resulting thumbnail image, defaults to a file named same as
+ *                                 the source file inside 'thumbnails' subdirectory of the source file directory.
+ * @param {number} width - Specifies the width of the thumbnail image, in pixels (default is 240).
+ * @param {number} height - Specifies the height of the thumbnail image, in pixels (default is 130).
+ */
```
Comment (VasilyStrelyaev) [inline src/api/generate-thumbnail.js]:
```
/**
 * Creates a thumbnail image from the specified PNG image file.
 * @function
 * @async
 * @name generateThumbnail
 * @param {string} sourcePath - Specifies the path to the source image in PNG format.
 * @param {string} thumbnailPath - Specifies the path to the resulting thumbnail image.
*                                 Defaults to '<sourcePathDirectory>/thumbnails/<sourcePathFileName>'
 * @param {number} width - Specifies the width of the thumbnail image, in pixels (default is 240).
 * @param {number} height - Specifies the height of the thumbnail image, in pixels (default is 130).
 */
```

Comment (VasilyStrelyaev) [inline src/api/generate-thumbnail.js]: Take a note: `imagePath` => `sourcePath`

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by AndreyBelym** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by AndreyBelym** [comment]
FPR, fixed code style

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by AndreyBelym** [comment]
Oh, stop, forgot to bump the version, please, hold on.

**Comment by AlexanderMoskovkin** [comment]
The `0.15.0` version is not publish yet. You should not increase the version to `0.16.0` here

**Comment by AndreyBelym** [comment]
Really, reverted back to `@0.15.0`.

**Pull request merged**